include/utils/Simplify.h

```diff
@@ -160,32 +182,41 @@ class Simplify
         };
         std::priority_queue<std::pair<size_t, coord_t>, std::vector<std::pair<size_t, coord_t>>, decltype(comparator)> by_importance(comparator);
 
-        //Add the initial points.
-        for(size_t i = 0; i < polygon.size(); ++i)
-        {
-            const coord_t vertex_importance = importance(polygon, to_delete, i, is_closed);
-            by_importance.emplace(i, vertex_importance);
-        }
-
-        //Iteratively remove the least important point until a threshold.
         Polygonal result = polygon; //Make a copy so that we can also shift vertices.
-        coord_t vertex_importance = 0;
-        while(by_importance.size() > min_size)
+        for (int64_t current_removed = -1; (polygon.size() - current_removed) > min_size && current_removed != 0;)
         {
-            std::pair<size_t, coord_t> vertex = by_importance.top();
-            by_importance.pop();
-            //The importance may have changed since this vertex was inserted. Re-compute it now.
-            //If it doesn't change, it's safe to process.
-            vertex_importance = importance(result, to_delete, vertex.first, is_closed);
-            if(vertex_importance != vertex.second)
+            current_removed = 0;
+
+            //Add the initial points.
+            for (size_t i = 0; i < result.size(); ++i)
             {
-                by_importance.emplace(vertex.first, vertex_importance); //Re-insert with updated importance.
-                continue;
+                if (to_delete[i])
+                {
+                    continue;
+                }
+                const coord_t vertex_importance = importance(result, to_delete, i, is_closed);
+                by_importance.emplace(i, vertex_importance);
             }
 
-            if(vertex_importance <= max_deviation * max_deviation)
+            //Iteratively remove the least important point until a threshold.
+            coord_t vertex_importance = 0;
+            while ((polygon.size() - current_removed) > min_size && ! by_importance.empty())
             {
-                remove(result, to_delete, vertex.first, vertex_importance, is_closed);
+                std::pair<size_t, coord_t> vertex = by_importance.top();
+                by_importance.pop();
+                //The importance may have changed since this vertex was inserted. Re-compute it now.
+                //If it doesn't change, it's safe to process.
+                vertex_importance = importance(result, to_delete, vertex.first, is_closed);
+                if (vertex_importance != vertex.second)
+                {
+                    by_importance.emplace(vertex.first, vertex_importance); //Re-insert with updated importance.
+                    continue;
+                }
+
+                if (vertex_importance <= max_deviation * max_deviation)
+                {
+                    current_removed += remove(result, to_delete, vertex.first, vertex_importance, is_closed) ? 1 : 0;
```

Comment:
```suggestion
                    auto removed = remove(result, to_delete, vertex.first, vertex_importance, is_closed);
                    if (removed)
                    {
                        current_removed +=  1;
                    }
```

Reply:
Not sure about this one. I like the compactness of the current code (too much lines is also less readable after all), but I'm not really against the change, because in a way it does make it more clear.